allennlp/tools/drop_eval.py

```diff
@@ -179,21 +185,23 @@ def evaluate_json(annotations: Dict[str, Any], predicted_answers: Dict[str, Any]
             max_f1_score = 0.0
             max_type = None
             if query_id in predicted_answers:
-                if "answer" in predicted_answers[query_id]:
-                    predicted = predicted_answers[query_id]["answer"]
-                else:
-                    predicted = predicted_answers[query_id]
+                predicted = predicted_answers[query_id]
+                candidate_answers = [qa_pair["answer"]]
+                if "validated_answers" in qa_pair and len(qa_pair["validated_answers"]) > 0:
+                    candidate_answers += qa_pair["validated_answers"]
+                for answer in candidate_answers:
+                    gold_answer, gold_type = answer_json_to_strings(answer)
+                    em_score, f1_score = get_metrics(predicted, gold_answer)
+                    if gold_answer[0].strip() != "":
+                        max_em_score = max(max_em_score, em_score)
+                        max_f1_score = max(max_f1_score, f1_score)
+                        if max_em_score == em_score or max_f1_score == f1_score:
+                            max_type = gold_type
             else:
                 print("Missing prediction for question: {}".format(query_id))
-                predicted = None
-            for answer in [qa_pair["answer"]] + qa_pair["validated_answers"]:
-                gold_answer, gold_type = answer_json_to_strings(answer)
-                em_score, f1_score = get_metrics(predicted, gold_answer)
-                if gold_answer[0].strip() != "":
-                    max_em_score = max(max_em_score, em_score)
-                    max_f1_score = max(max_f1_score, f1_score)
-                    if max_em_score == em_score or max_f1_score == f1_score:
-                        max_type = gold_type
+                _, max_type = answer_json_to_strings(qa_pair["answer"])
```

Comment:
It looks like mypy is complaining about re-assigning a different type to `_` :(.  If you call this `__` instead mypy should stop complaining.